File: Node/src/test/integration/vn/MerchantIntegrationOffline.test.ts
import { MerchantIntegrationOffline } from '../../../MerchantIntegrationOffline';
import * as Utils from '../../../utils/Utils';

const env = 'STG';
const country = 'VN';
const partnerId = process.env.VN_STG_POS_PARTNER_ID!;
const partnerSecret = process.env.VN_STG_POS_PARTNER_SECRET!;
const merchantId = process.env.VN_STG_POS_MERCHANT_ID!;
const terminalId = process.env.VN_STG_POS_TERMINAL_ID!;

describe('Test VN MerchantIntegrationOffline', () => {
  let client: MerchantIntegrationOffline;
  let partnerTxID: string;
  let partnerGroupTxID: string;
  let msgID: string;
  let origPartnerTxID: string;
  beforeAll(() => {
    client = new MerchantIntegrationOffline({ env, country, partnerId, partnerSecret, merchantId, terminalId });
  });

  beforeEach(() => {
    partnerTxID = Utils.generateRandomString(32);
    partnerGroupTxID = Utils.generateRandomString(32);
    msgID = Utils.generateMsgId();
    origPartnerTxID = Utils.generateRandomString(32);
  });

  it('Test createQrCode', async () => {
    const amount = 2000;
    const currency = 'VND';

    const response = await client.createQrCode({ msgID, partnerTxID, amount, currency });
    const { status, data } = response;
    expect(status).toBe(200);

    expect(data.msgID).toEqual(msgID);
    expect(data.qrcode).isPrototypeOf(String);
    expect(data.txID).isPrototypeOf(String);
  });
});

File: Go/test/integration/MerchantIntegrationOffline_test.go
package integration

import (
	"fmt"
	"io"
	"os"
	"testing"

	"github.com/grab/grabpay-merchant-sdk/dto"
	"github.com/grab/grabpay-merchant-sdk/merchant"
	"github.com/grab/grabpay-merchant-sdk/utils"
)

var (
	_env           = "STG"
	_country       = "VN"
	_partnerID     = os.Getenv("VN_STG_POS_PARTNER_ID")
	_partnerSecret = os.Getenv("VN_STG_POS_PARTNER_SECRET")
	_merchantID    = os.Getenv("VN_STG_POS_MERCHANT_ID")
	terminalID     = os.Getenv("VN_STG_POS_TERMINAL_ID")
)

func TestCreateQRCode(t *testing.T) {
	merchant := merchant.NewMerchantOffline(_env, _country, _partnerID, _partnerSecret, _merchantID, terminalID)
	params := new(dto.PosCreateQRCodeParams)
	params.PartnerTxID = utils.GenerateRandomString(32)
	params.Amount = int64(10000)
	params.MsgID = utils.GenerateMD5(utils.GenerateRandomString(32))
	params.Currency = "VND"
	resp, err := merchant.PosCreateQRCode(nil, params)

	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()
	bodyBytes, err := io.ReadAll(resp.Body)
	fmt.Println(string(bodyBytes))

}
